### PR TITLE
add bulk results submissions tab

### LINKED ISSUE #3645: Add bulk results submission tab to SR
PART of https://github.com/CDCgov/prime-simplereport/issues/3627

### Description
Add UI to view csv upload submissions to RS directly from SimpleReport

### Proposed solution 
**Retrieve submissions:** currently RS UI uses the following api endpoint GET api/waters/org/{orgName}/submissions, this works for them, but for us, it will not since every submission will be under the SimpleReport Org as far as RS is concerned. so we would want a way to link to it, a proposal would be to create an UploadsTable in out db and each entry will contain the report id from RS. Retrieving submissions will be loading the contents of that UploadsTable for that org.

Things todo in this ticket
- Bring the submission tab ui from RS 
- Use the `UploadsTable` to retrieve submitted results for the logged in org
- The ui should be be accessible through `simplereport.gov/app/results/upload/submissions`
- note that clicking on individual report id will be part of a [this ticket](https://github.com/CDCgov/prime-simplereport/issues/3646)
- the ui should only be accessible to user with `SR-CSV-UPLOADER-PILOT` permission (okta group), otherwise route to home dashboard

### Additional context
image

You should all be able to login to the RS staging account via your OktaPreview credentials via https://staging.reportstream.cdc.gov/login

use **#prime-reportstream-feature-port** slack channel
[Figma](https://www.figma.com/proto/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?page-id=8979%3A27220&node-id=8979%3A27225&viewport=260%2C48%2C0.55&scaling=min-zoom&starting-point-node-id=8979%3A27225)
RS UI [source code](https://github.com/CDCgov/prime-reportstream/tree/master/frontend-react/src/pages)
[Spec Doc](https://cdc.sharepoint.com/:w:/t/USDSatCDC/EZPs906A_VlHreQdunv0NSoBI7Kxo3_bmn8DJD2AKQ_QeQ?e=JwyL2A)
Designs to follow.

### PR DESCRIPTION
**Pull request by zdeveloper**
# FRONTEND PULL REQUEST

## Related Issue

- resolves #3645 

## Changes Proposed

- Add Submissions tab

## Testing

- you can access this page using `/results/upload/submissions` path
- this is hidden behind a feature flag
 
## Screenshots / Demos

image

### Design
- [X] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [X] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [X] Any content changes have been approved by content team

### Support
- [X] Any changes that might generate new support requests have been flagged to the support team
- [X] Any changes to support infrastructure have been demo'd to support team

### Security
- [X] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [X] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

File: frontend/src/app/testResults/submissions/Submissions.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import { useState } from "react";
+
+import Pagination from "../../commonComponents/Pagination";
+import { formatDateWithTimeOption } from "../../utils/date";
+import {
+  GetUploadSubmissionsQuery,
+  useGetUploadSubmissionsQuery,
+} from "../../../generated/graphql";
+import {
+  // setEndTimeForDateRange,
```
Comment (emmastephenson) [inline frontend/src/app/testResults/submissions/Submissions.tsx:13]: obsolete comment?

File: frontend/src/app/testResults/submissions/Submissions.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    fetchPolicy: "no-cache",
+    variables: {
+      pageSize,
+      pageNumber: pageNumber - 1,
+      startDate,
+      endDate,
+    },
+  });
+
+  console.log(submissions);
```
Comment (emmastephenson) [inline frontend/src/app/testResults/submissions/Submissions.tsx:37]: Can you take out this debugging statement?

Comment (zdeveloper) [reply frontend/src/app/testResults/submissions/Submissions.tsx:37]: lol thanks for catching this!

File: frontend/src/app/testResults/submissions/Submissions.tsx (hunk 1 above)
Comment (zdeveloper) [reply frontend/src/app/testResults/submissions/Submissions.tsx:13]: nice catch! I used it and then decided not to use it

**Review by emmastephenson — APPROVED** [review]
Overall this looks great! I played around with it on test and everything seems to be WAI.

One thing to potentially address in a later PR is that when filters are applied, the entire component reloads. It'd be nice if this behaved more like the existing results page, where only the list of results reloads on filter application.

**Review by emmastephenson — APPROVED** [review]

**Review by nathancrtr — APPROVED** [review]
Looks awesome! 😎 thanks for the thorough tests

**Pull request merged**